File: Progetto_JAVA/src/progetto/Automobile.java
package progetto;

//Classe Automobile, sottoclasse di Veicolo
public final class Automobile extends Veicolo implements Printable {
	String marca;
	String modello;
	/**
	 * 
	 * @param p PROPRIETARIO
	 * @param t TARGA
	 * @param l LUNGHEZZA
	 * @param a ALTEZZA
	 */
	public Automobile(String p, String t, int l, int a, String ma, String mo) {
		super(p, t, l, a);
		this.marca=ma;
		this.modello=mo;
	}

	public void printInfo() {
		System.out.println("     Dati di " + this.targa + " :");
		System.out.print("     Tipologia: Automobile, ");
		System.out.print("Proprietario: " + this.proprietario);
		System.out.print(", Lunghezza: " + this.lunghezza);
		System.out.print(", Altezza: " + this.altezza);
		System.out.print(", Marca: " + this.marca);
		System.out.print(", Modello: " + this.modello);
		System.out.println("");
	}

	public String getMarca() {
		return marca;
	}

	public String getModello() {
		return modello;
	}

	@Override
	public String toString() {
		return "Automobile [marca=" + marca + ", modello=" + modello + ", proprietario=" + proprietario + ", targa="
				+ targa + ", lunghezza=" + lunghezza + ", altezza=" + altezza + "]";
	}
	

}

File: Progetto_C++_TESTA2/src/Veicoli/Automobile.cpp
/*
 * Automobile.cpp
 *
 *  Created on: 25 set 2023
 *      Author: Testa Lorenzo
 */

#include "Automobile.h"
#include <sstream>
#include <iostream>
#include <string.h>
using namespace std;

Automobile::Automobile(string p, string t, int l = 500, int a = 200, string ma =
		"", string mo = "") :
		Veicolo(p, t, l, a) {
	Marca = ma;
	Modello = mo;
}

Automobile::~Automobile() {
}

Automobile::Automobile(const Automobile &other) :
		Veicolo(other.Proprietario, other.Targa, other.Lunghezza, other.Altezza) {
	this->Marca = other.Marca;
	this->Modello = other.Modello;
}

inline void Automobile::printInfo() {
	cout << "Targa: " << this->Targa << " Proprietario: " << this->Proprietario
			<< endl << "Marca: " << this->Marca << " Modello: " << this->Modello
			<< endl << "Lunghezza: " << this->Lunghezza << " Altezza: "
			<< this->Altezza << endl;
}

inline const string Automobile::getMarca() {
	return Marca;
}
inline const string Automobile::getModello() {
	return Modello;
}

const string Automobile::getTOString() {
	stringstream streamer;
	streamer << Veicolo::getTOString() << " Marca: " << this->Marca
			<< " Modello: " << this->Modello << endl;
	return streamer.str();
}
